pkg/instancetype/instancetype.go

```diff
@@ -750,6 +750,10 @@ func (m *InstancetypeMethods) InferDefaultInstancetype(vm *virtv1.VirtualMachine
 		return nil, err
 	}
 
+	if ignoreFailure {
+		vm.Spec.Template.Spec.Domain.Memory = nil
```

Comment:
This works because we mutate the same `VirtualMachine` object that's later used to build patches in the mutation webhook [1] . A code comment might be useful here and tbh I don't recall why I'm even bothering to return the `Matchers` in these mutation functions.

[1] 
https://github.com/kubevirt/kubevirt/blob/5b61a18f4c3b6c549d50fd201664c21720e3dbe2/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go#L127-L138

Reply:
I reworked the PR to mutate the VM in-place instead of returning matchers.